SEP-0012.md

```diff
@@ -0,0 +1,553 @@
+
+
+## Overview
+
+Before jumping into the details of the API, this section will provide a *brief* high level summary of the way the following three classes interact.
+For a detailed introduction to the whole `ndcube` package please see the [ndcube documentation](https://docs.sunpy.org/projects/ndcube/en/v2.0.0b2/introduction.html), a high level familiarity with the package is assumed in the rest of this SEP.
+
+The `NDCube` object inherits from `astropy.nddata.NDData` and is a container which holds an array, a WCS and extra information about the array like a mask, unit, uncertainty etc.
+One of the main bits of functionality that `NDCube` adds on top of `NDData` is some API which allows the user to add "extra" coordinate information to describe the array.
+This information is stored in the `ExtraCoords` object which is accessible from the `NDCube.extra_coords` property.
+The main use case for these "extra" coords is when an axis of the array has one coordinate described by the primary WCS but that axis also has a secondary coordinate.
+A concrete example of this in solar physics is a rastering spectrograph, where one of the spatial axes (the array axis along which the raster is built) also has a time coordinate, this coordinate is not normally included in the WCS.
+
+The second added component, new in 2.0, is the `GlobalCoords` class.
+This class, in the same way as `ExtraCoords` holds additional coordinate information, however, the main difference is that `GlobalCoords` are not associated with an axes of the array.
+This, almost always, means they are scalars; for example a point in space or a time.
+This `gobal_coords` attribute could hold the HPC coordinate of a line spectra, or the time of a map in a way which allows predictable programmatic access to this metadata.
+The other thing which is implemented in ndcube, is that when the array is slice such that a dimension is dropped, the coordinate from that axis of the WCS is added automatically to `GlobalCoords`, meaning this metadata is not lost.
+
+
+### On WCSes
+
+The API specification below includes three distinct WCS like objects (all conforming to the Astropy WCS API).
+These are `.wcs`, `.extra_coords` and `.combined_wcs`.
+The `.wcs` attribute is the main WCS object for the array; it is required that it have the same number of pixel dimensions as there are array dimensions, the number of world dimensions is arbitrary.
+The `.extra_coords` object implements the WCS API but is not required to be a valid WCS for the whole array; the number of pixel dimensions do not have to match the number of array dimensions.
+This is because the user can add as many coordinates to the extra coords as is needed, the `ExtraCoords` object then keeps a mapping of the pixel dimensions in it's WCS to the array dimensions in the `NDCube` object.
```

Comment:
```suggestion
This is because the user can add as many coordinates to the extra coords as is needed, the `ExtraCoords` object then keeps a mapping of the pixel dimensions in its WCS to the array dimensions in the `NDCube` object.
```